actions_helper/main.py

```diff
@@ -1,72 +1,102 @@
 
 
 @click.group()
 def cli():
     pass
 
 
-@cli.command(name="wait-for-service-stability")
-@click.option("--aws-region", default=os.environ.get("AWS_DEFAULT_REGION"), type=str)
-@click.option("--cluster", type=str)
-@click.option("--service", type=str)
-def cmd_wait_for_service_stability(
+@cli.command(
+    name="ecs-deploy",
+    short_help="Deploy production image to AWS ECS",
+)
+@click.option("--environment", type=click.Choice(["dev", "test", "live"]))
+@click.option("--ecr-repository", envvar="ECR_REPOSITORY", type=str)
+@click.option("--deployment-tag", envvar="DEPLOYMENT_TAG", type=str)
+@click.option("--image-tag", envvar="IMAGE_TAG", type=str)
+@click.option("--run-preflight", envvar="RUN_PREFLIGHT", type=bool)
+@click.option("--desired-count", default=2, type=int)
+@click.option("--aws-region", envvar="AWS_DEFAULT_REGION", type=str)
+def cmd_ecs_deploy(
+    environment: str,
+    ecr_repository: str,
+    deployment_tag: str,
+    image_tag: str,
+    run_preflight: bool,
+    desired_count: str,
     aws_region: str,
-    cluster: str,
-    service: str,
 ):
-    wait_for_service_stability(
-        ecs_client=boto3.Session(region_name=aws_region).client("ecs"),
-        cluster=cluster,
-        service=service,
-    )
+    ecs_client = boto3.Session(region_name=aws_region).client("ecs")
+    ecr_client = boto3.Session(region_name=aws_region).client("ecr")
+    production_task_definition = local_task_definition = preflight_output = None
+    try:
+        click.echo("Getting docker image URI...")
+        image_uri = get_image_uri(ecr_client=ecr_client, ecr_repository=ecr_repository, tag=image_tag)
 
+        click.echo("Creating local task definition...")
+        local_task_definition = create_task_definition(
+            ecs_client=ecs_client,
+            application_id=f"{ecr_repository}-local-exec-{environment}",
+            deployment_tag=deployment_tag,
+            image_uri=image_uri,
+        )
 
-@cli.command(name="wait-for-task-stopped")
-@click.option("--aws-region", default=os.environ.get("AWS_DEFAULT_REGION"), type=str)
-@click.option("--cluster", type=str)
-@click.option("--task", type=str)
-def cmd_wait_for_task_stopped(
-    aws_region: str,
-    cluster: str,
-    task: str,
-):
-    wait_for_task_stopped(
-        ecs_client=boto3.Session(region_name=aws_region).client("ecs"),
-        cluster=cluster,
-        task=task,
-    )
+        click.echo("Creating production task definition...")
+        production_task_definition = create_task_definition(
+            ecs_client=ecs_client,
+            application_id=f"{ecr_repository}-{environment}",
+            deployment_tag=deployment_tag,
+            image_uri=image_uri,
+        )
 
+        if run_preflight:
+            click.echo("Run preflight enabled")
+            preflight_output = run_preflight_container(
+                ecs_client=ecs_client,
+                service=f"{ecr_repository}-{environment}",
```

Comment:
This line gets repeated multiple times in this file, should we introduce a variable for the service name?